### PR TITLE
Self-karma positive or negative messaging

### LINKED ISSUE #5: Scold folks who try to give themselves points with configurable warning.
Currently, if someone attempts to give themselves points, it is silently ignored.

Desired, the user should be scolded with configured warning. If no warning is configured, it should default to the Hipchat Karma bot version: "Don't be a weasel."

### PR DESCRIPTION
**Pull request by diachini**
## Description
- Scold a user for attempting to give themself karma
- Encourage a user who attempted to remove karma from themself

## QA
- [x] Confirm default self-plus works:
```
jetbot> shell++
jetbot> Did you really just try to give yourself karma?  REALLY?!
```
- [x] Confirm default self-minus works:
```
jetbot> shell--
jetbot> There there, buddy... it's gonna be all right.
```

- [x] `SELF_KARMA_SCOLDING='For shame.' HUMILITY_SCOLDING="You're FINE" bin/hubot`
- [x] Confirm custom messaging works:
```
jetbot> shell++
jetbot> For shame.
jetbot> shell--
jetbot> You're FINE
```

## Related Issue
closes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ API
 * `thing--` - remove a point from `thing`
 * `--` - remove a point from the most previously voted-on thing
 * `thing-- for stuff` - keep track of why you removed thing points
+* `@me++` - get scolded (via configurable `SELF_KARMA_SCOLDING` message) for attempting to add a point to oneself
```
Comment (regentgal) [inline README.md:24]: will be self explanatory. users won't understand the constant references.

File: src/mega-plusplus.coffee [hunk 2]
```diff
@@ -43,6 +43,8 @@ module.exports = (robot) ->
   reasonsKeyword = process.env.HUBOT_PLUSPLUS_REASONS or 'raisins'
   maxPoints = process.env.HUBOT_PLUSPLUS_MAX_POINTS or 5
   reasonConjunctions = process.env.HUBOT_PLUSPLUS_CONJUNCTIONS or 'for|because|cause|cuz|as'
+  selfKarmaScolding = process.env.SELF_KARMA_SCOLDING or 'Did you really just try to give yourself karma?  REALLY?!'
+  humilityScolding = process.env.HUMILITY_SCOLDING or 'There there, buddy... it\'s gonna be all right.'
```
Comment (regentgal) [inline src/mega-plusplus.coffee:47]: lets make these names consistent - keep `selfKarmaScold` for self++ and `selfKarmaSupport`

File: src/mega-plusplus.coffee [hunk 3]
```diff
@@ -79,7 +81,8 @@ module.exports = (robot) ->
     # do the {up, down}vote, and figure out what the new score is
     requestedMagnitude = (operator.length - 1)
     magnitude = Math.min(requestedMagnitude, maxPoints)
-    [score, reasonScore] = if operator.charAt(0) == "+"
+    wasAPositiveChange = operator.charAt(0) == "+"
```
Comment (regentgal) [inline src/mega-plusplus.coffee:84]: `isPositiveKarma`

**Comment by regentgal** [comment]
#13 introduces a changelog. please work with @Zajn so you can also had an entry for this as well.

**Comment by diachini** [comment]
Just like #8 - this was a piece of functionality that we had in Hipchat that I don't think I miss all that much. Closing to stop the daily GitHub reminder since I don't foresee actually finishing this up.

**Pull request closed without merging**